mlflow/store/artifact/databricks_artifact_repo.py

```diff
@@ -320,5 +339,88 @@ def _download_file(self, remote_file_path, local_path):
         read_credentials = self._get_read_credentials(self.run_id, run_relative_remote_file_path)
         self._download_from_cloud(read_credentials.credentials, local_path)
 
+    def download_artifacts(self, artifact_path, dst_path=None):
+        """
+        Parallelized implementation of `download_artifacts` for Databricks.
+        """
+
+        def download_file(fullpath):
+            """
+            Submit a download of `fullpath` to the thread pool and return the
+            resultant Future.
```

Comment:
It looks the actual return is a tuple of downloaded local file path and its corresponding future.
In fact if a dict of {file_path: future} is returned here, the return signature will be the same for both download_file() and download_artifact_dir(). The handling of futures can be combined then (see below comment).

Reply:
I like this idea! In addition to a collection of futures, we also need to maintain the top-level destination directory that should be returned by `download_artifacts()`, hence the first element of the tuple.

To achieve consistent return types and deduplicate the logic, I've modified the signature of each method to return a documented `InflightDownload` named tuple whose first element is the top-level destination directory and whose second element is the mapping you outlined above.

This mapping you suggested has the added benefit of enabling us to provide failure information for each specific path in the event of one or more errors.